RUMM-1939 Document data storage regarding encryption (#756)
DataDog/dd-sdk-ios: docs/log_collection.md

- if changed to `.granted`, the SDK will send all current and future data to Datadog;
- if changed to `.notGranted`, the SDK will wipe all current data and will not collect any future data.

> > Before uploading to Datadog, data is stored in cleartext in the cache directory (`Library/Caches`) of your application bundle.

Reviewer: Any reason for using quotation? I think it should be just a plain text:
```suggestion
Before uploading to Datadog, data is stored in cleartext in the cache directory (`Library/Caches`) of your application bundle.
```

Author: Not really, it was more about rendering it as a note.

> When writing your application, you can enable development logs. All internal messages in the SDK with a priority equal to or higher than the provided level are then logged to console logs.

Reviewer 2: ```suggestion
When writing your application, enable development logs to log to console all internal messages in the SDK with a priority equal to or higher than the provided level.
```
